examples/embassy-http-server/src/main.rs

```diff
@@ -87,68 +85,49 @@ async fn web_task(
     }
 }
 
-struct WebServer {
+// TODO: macro up this up
+use riot_rs::embassy::{arch::OptionalPeripherals, Spawner};
+#[riot_rs::embassy::distributed_slice(riot_rs::embassy::EMBASSY_TASKS)]
+#[linkme(crate = riot_rs::embassy::linkme)]
+fn web_server_init(spawner: &Spawner, peripherals: &mut OptionalPeripherals) {
```

Comment:
So here, b/c the Buttons peripherals are optional, I found it easier to just write a custom non-async init function.